alchemy/src/supabase/project.ts

```diff
@@ -0,0 +1,267 @@
+import type { Context } from "../context.ts";
+import {
+  Resource,
+  ResourceKind,
+  ResourceID,
+  ResourceFQN,
+  ResourceScope,
+  ResourceSeq,
+} from "../resource.ts";
+import { Scope } from "../scope.ts";
+import {
+  createSupabaseApi,
+  type SupabaseApiOptions,
+  type SupabaseApi,
+} from "./api.ts";
+import { handleApiError } from "./api-error.ts";
+
+/**
+ * Properties for creating or updating a Supabase Project
+ */
+export interface ProjectProps extends SupabaseApiOptions {
+  /**
+   * Name of the project (optional, defaults to resource ID)
+   */
+  name?: string;
+
+  /**
+   * ID of the organization that will own this project
+   */
+  organizationId: string;
+
+  /**
+   * Region where the project will be hosted
+   */
+  region: string;
+
+  /**
+   * Database password for the project
+   */
+  dbPass: string;
+
+  /**
+   * Desired instance size for the project
+   */
+  desiredInstanceSize?: string;
+
+  /**
+   * Template URL for project initialization
+   */
+  templateUrl?: string;
+
+  /**
+   * Whether to adopt an existing project instead of failing on conflict
+   */
+  adopt?: boolean;
+
+  /**
+   * Whether to delete the project on resource destruction
+   */
+  delete?: boolean;
+}
+
+/**
+ * Supabase Project resource
+ */
+export interface ProjectResource extends Resource<"supabase::Project"> {
```

Comment:
The name should be Project. 

We don't add the Resource suffix. Apply this universally across all the new resources.